src/VisualStudio/IntegrationTest/TestUtilities/InProcess/SolutionExplorer_InProc.cs

```diff
@@ -214,6 +215,38 @@ public void AddReference(string projectName, string fullyQualifiedAssemblyName)
             ((VSProject)project.Object).References.Add(fullyQualifiedAssemblyName);
         }
 
+        public void AddPackageReference(string projectName, string packageName, string version)
+        {
+            var project = GetProject(projectName);
+
+            if (project.Object is IVsBrowseObjectContext browseObjectContext)
```

Comment:
Can we throw if `project.Object` is not of type `IVsBrowseObjectContext` instead of silently failing?